distributed/worker_state_machine.py

```diff
@@ -3037,9 +3036,15 @@ def _validate_task_executing(self, ts: TaskState) -> None:
             assert dep.key in self.data or dep.key in self.actors
 
     def _validate_task_ready(self, ts: TaskState) -> None:
-        assert ts.key in pluck(1, self.ready)
+        if ts.state == "ready":
+            assert ts.key in pluck(1, self.ready)
```

Comment:
```suggestion
            assert ts.key in pluck(1, self.ready)
            assert ts.key not in self.constrained
```

Reply:
Superseded by suggestions in https://github.com/dask/distributed/pull/6698#pullrequestreview-1034032402